### PR TITLE
Admin order payments ui cleanup

### PR DESCRIPTION
**Pull request by tvdeyen**
Extracted from #2255 

Made some changes to the admin order payments UI:

- Fixes the gateway payment method form
- Make the number with currency component suitable for smaller widths
- Re-arrange the order payments table

### Screenshots

localhost_3000_admin_orders_r123456789_payments_new laptop with hidpi screen 1

localhost_3000_admin_orders_r123456789_payments_new laptop with hidpi screen

payments - before

payments - after

### CODE REVIEW

File: backend/app/views/spree/admin/payments/_list.html.erb [hunk 1]
```diff
@@ -1,31 +1,47 @@
...
             <%= t(payment.state, scope: 'spree.payment_states') %>
           </span>
         </td>
+        <td class="amount align-right">
+          <div class="editing-show">
+            <div class="input-group">
+              <div class="input-group-addon number-with-currency-symbol">
+                <%= ::Money::Currency.find(@order.currency).symbol %>
+              </div>
+              <%= number_field_tag :amount, payment.amount, class: 'js-edit-amount align-right form-control', step: :any %>
```
Comment (kennyadsl) [inline backend/app/views/spree/admin/payments/_list.html.erb]: We decided to [use "text" input for currency fields](https://github.com/solidusio/solidus/pull/2232), right?

Comment (tvdeyen) [reply backend/app/views/spree/admin/payments/_list.html.erb]: @kennyadsl Fixed it

File: backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Spree.Views.Payment.PaymentRow = Backbone.View.extend
```
Comment (ccarruitero) [inline backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee]: I think the file should be in javascript instead coffee.

From [backend README](https://github.com/solidusio/solidus/tree/master/backend):

>Though we have existing CoffeeScript files, any new files should be in javascript (ES5).

Comment (kennyadsl) [reply backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee]: Ah, you are right, good catch!

File: backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  events:
+    "click .js-edit": "onEdit"
+    "click .js-save": "onSave"
+    "click .js-cancel": "onCancel"
+
+  onEdit: (e) ->
+    e.preventDefault()
+    @$el.addClass("editing")
```
Comment (ccarruitero) [inline backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee]: Surely is outside the scope of this changes.

But we have similar backbone views that use a conditional value `editing` with a handlebars template.  (ex: [orders/shipping_methods](https://github.com/solidusio/solidus/blob/926ea6880b232b353d8cea1479a2f1bcc41ac279/backend/app/assets/javascripts/spree/backend/views/order/shipping_method.js#L18)) I think that approach is better than show/hide with css.

Maybe we can create a common view for use in this cases.

File: backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee (hunk 2 above)
Comment (tvdeyen) [reply backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee]: Will do, thanks!

File: backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee (hunk 3 above)
Comment (tvdeyen) [reply backend/app/assets/javascripts/spree/backend/views/payment/payment_row.js.coffee]: Absolutely. Will see if I can sneak this into this PR

**Comment by tvdeyen** [comment]
@kennyadsl made the requested changes

**Pull request merged**